samples/v2023-09/job_templates/bash-in-docker.yaml

```
@@ -0,0 +1,104 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# ----
+# Demonstrates
+# ----
+# This demonstrates a way that Tasks can be run inside of a docker container.
+#
+# Run with:
+#  openjd run bash-in-docker.yaml --step RunInDocker
+#
+# ----
+# Requirements
+# ----
+# - linux (tested on Amazon Linux 2)
+# - docker (tested wth version 20.10.25)
+# - openssl (tested with version 1.0.2k-fips)
+#
+# -----
+# Contributors to this template:
+#    Daniel Neilson (https://github.com/ddneilson)
+
+name: DockerExample
+specificationVersion: jobtemplate-2023-09
+
+steps:
+- name: RunInDocker
+
+  parameterSpace:
+    taskParameterDefinitions:
+    - name: Phrase
+      type: STRING
+      range:
+      - "Hello"
+      - "Greetings"
+      - "Hi"
+
+  stepEnvironments:
+  - name: DockerContainer
+    script:
+      actions:
+        onEnter:
+          command: bash
+          args: ["{{Env.File.Enter}}"]
+        onExit:
+          command: bash
+          args: ["{{Env.File.Exit}}"]
+      embeddedFiles:
+      - name: Enter
+        type: TEXT
+        data: |
+          #!/bin/bash
+
+          # Any command failure in the script fails the whole thing
+          set -xeou pipefail
+
+          # Start the docker container running in the background with a randomly determined container name
+          # The random name enabled running multiple of this same Job on the same host at the same time.
+          CONTAINER_NAME="container-$(openssl rand -hex 16)"
+          echo $CONTAINER_NAME > .docker_container_name.txt
+          nohup docker run --rm --name $CONTAINER_NAME ubuntu:24.04  /bin/bash -c "while true; do sleep 10; done" > container.out 2>&1&
+
+          # Give the container a moment to actually start
+          sleep 2
```

We can probably simplify this using the [`-d` (detached mode)](https://docs.docker.com/reference/cli/docker/container/run/#detach):

```suggestion
         docker run --rm -d --name $CONTAINER_NAME ubuntu:24.04  /bin/bash -c "while true; do sleep 10; done"
```

---

Right you are. Works perfectly! Thanks